docs/release_process.rst

```diff
@@ -15,79 +14,112 @@ number occurs.
 Create and Merge Version Update Branch
 --------------------------------------
 
-Patch Release
-^^^^^^^^^^^^^
-A patch release should only contain fixes for bugs or issues with an existing
-release.
-No new features or functionality should be introduced in a patch release.
-As such, patch releases should only ever be based on an existing release point.
+Major and Minor Releases
+^^^^^^^^^^^^^^^^^^^^^^^^
+Major and minor releases may add one or more trivial or non-trivial features
+and functionalities.
 
-1. Create a new branch off of the appropriate ``vX.Y.Z`` tag named something
-   like ``release-patch-{NEW_VERSION}``, where ``NEW_VERSION`` is an increase
-   in the ``Z`` version component.
+1. Create a new branch off of the ``master`` named something like
+   ``update-to-v{NEW_VERSION}``, where ``NEW_VERSION`` is the new ``X.Y``
+   version.
 
-   a. Use ``poetry version patch`` to increase the patch value appropriately in
-      the :file:`pyproject.toml` file.
+   a. Use the ``scripts/update_release_notes.sh`` script to update the project
+      version number, create ``docs/release_notes/v{NEW_VERSION}.rst``, and add a
+      new pending release notes stub file.
 
-   b. Rename the ``docs/release_notes/pending_patch.rst`` file to
-      ``docs/release_notes/v{VERSION}.rst``, matching the new version value.
-      Add a descriptive paragraph under the title section summarizing this
-      release.
+      .. code-block:: bash
 
-   c. Add a reference to the new release notes RST file in
-      ``docs/release_notes.rst``.
+         $ ./scripts/update_release_notes.sh minor
```

Comment:
Is it clear that this script works also for major releases, by simply changing the argument passed from `minor` to `major`?

Reply:
More so the latter, that it is clear here in the documentation that the script can be used for major, minor, and/or patch releases. It's not a major issue, so feel free to ignore (was mainly thinking about the failure mode where someone blindly copies the command).

Reply:
Valid, I can expand the example.